includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -234,7 +237,7 @@ public function get_scripts() {
 	 * @param array $attr_spec_list Attribute spec list.
 	 * @return array Modified attribute spec list.
 	 */
-	private function process_alternate_names( $attr_spec_list ) {
+	private function process_alternate_names( array $attr_spec_list ) {
```

Comment:
I strongly suggest not using actual typehints for `array`, as they introduce all sorts of problems with PHP. The main issue that any object that behaves like an array will not pass this type check, so you're forced to stay with dumb arrays and can never add any optimization later on.
```suggestion
	private function process_alternate_names( $attr_spec_list ) {
```

Reply:
That being said, this is a private method, so there's no problem in changing the typehint later if needed. However, it could just as well happen that the method is turned public later without reconsidering the `array` typehint, that's why I just generally avoid it.